### PR TITLE
[[ Bug 21570 ]] Tree View Widget - select new elements automatically

### PR DESCRIPTION
**Pull request by bwmilby**
The proposed change is the add a new property to the tree view widget `hiliteNewElement` which when true will cause the widget to automatically highlight the added key.  That will yield 2 messaged - `dataChanged` and `hiliteChanged`.

### CODE REVIEW

**Comment by montegoulding** [comment]
@bwmilby I'm wondering about ensuring the highlighted node is visible if there is automatic highlighting

**Comment by bwmilby** [comment]
> @bwmilby I'm wondering about ensuring the highlighted node is visible if there is automatic highlighting

If you `set the hilitedElement` to an node that is not visible, it currently does not scroll into view.  If you then scroll the view, it does properly show the highlight.  Adjusting the behavior should not be difficult, but I think it should be consistent in both cases.

**Comment by montegoulding** [comment]
Hmm... what about a property to control it. `scrollHilitedElementIntoView` or something very camel casey like that ;-) It seems to me likely that most people would want it to scroll into view

**Comment by bwmilby** [comment]
Sounds good.  I should be able to have something this weekend.

**Comment by bwmilby** [comment]
@montegoulding Here's a demo of how the new property impacts the widget.

treeviewtest2

treeviewtest

**Comment by montegoulding** [comment]
This seems great @bwmilby !

File: extensions/widgets/treeview/notes/feature-scroll_into_view.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Properties
+
+The tree view widget now has the ability to scroll the selected row into
```
Comment (montegoulding) [inline extensions/widgets/treeview/notes/feature-scroll_into_view.md]: @bwmilby I think you should discuss the new property `hiliteNewElement` here also.

Comment (bwmilby) [reply extensions/widgets/treeview/notes/feature-scroll_into_view.md]: Added property to the note.

**Comment by montegoulding** [comment]
@livecode-vulcan review ok b7e4a49

**Comment by livecode-vulcan** [comment]
💙 review by @montegoulding ok b7e4a4945574d4a4a697f9cb34235aec663be008

**Comment by livecode-vulcan** [comment]
😞 test failure b7e4a4945574d4a4a697f9cb34235aec663be008

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/93)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/85)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/85)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/85)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/85)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1607)
* try-community-universal-ios-iphoneos11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.4/builds/134)
* try-community-universal-ios-iphonesimulator11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.4/builds/134)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1637)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1630)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1627)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1663)
* try-community-x86_64-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1604)

**Comment by montegoulding** [comment]
@livecode-vulcan retry b7e4a49

**Comment by livecode-vulcan** [comment]
😎 test success b7e4a4945574d4a4a697f9cb34235aec663be008

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/94)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/86)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/86)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/86)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/86)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1608)
* try-community-universal-ios-iphoneos11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.4/builds/135)
* try-community-universal-ios-iphonesimulator11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.4/builds/135)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1638)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1631)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1628)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1664)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1605)

**Pull request merged**